src/VCR/Storage/AbstractStorage.php

```diff
@@ -69,14 +71,15 @@ public function __construct($cassettePath, $cassetteName, $defaultContent = '[]'
 
         if (!file_exists($file) || 0 === filesize($file)) {
             file_put_contents($file, $defaultContent);
-
             $this->isNew = true;
+        } else {
+            Assertion::file($file, "Specified path '{$file}' is not a file.");
+            Assertion::readable($file, "Specified file '{$file}' must be readable.");
+            if ($writeable) {
+                Assertion::writeable($file, "Specified path '{$file}' must be writeable.");
+            }
```

Comment:
no need to do these checks if we just created the file - obviously it was writeable and readable or file_put_contents would have failed.